saleor/graphql/attribute/utils.py

```diff
@@ -531,6 +571,20 @@ def validate_standard_attributes_input(
             )
 
 
+def validate_text_attributes_input(
+    attribute: "Attribute",
+    attr_values: "AttrValuesInput",
+    attribute_errors: T_ERROR_DICT,
+    variant_validation: bool,
+):
+    name_field = attribute.values.model.name.field  # type: ignore
+    attribute_id = attr_values.global_id
+    text = attr_values.values[0] if attr_values.values else ""
+
+    if len(text) > name_field.max_length:
+        attribute_errors[AttributeInputErrors.ERROR_MAX_LENGTH].append(attribute_id)
+
+
```

Comment:
I think, that for `text` attribute validation you should use `validate_standard_attributes_input` and add there this new validation, as it could also regard another attribute, and validations from `validate_standard_attributes_input` will be also useful with `text` attributes.